### PR TITLE
Adds new image metric - `ERGAS`

### PR DESCRIPTION
**Pull request by vumichien**
## What does this PR do?

Adds new Metric ERGAS #799 

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/image/ergas.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    References: Qian Du; Nicholas H. Younan; Roger King; Vijay P. Shah, "On the Performance Evaluation of
+    Pan-Sharpening Techniques" in IEEE Geoscience and Remote Sensing Letters, vol. 4, no. 4, pp. 518-522,
+    15 October 2007, doi: 10.1109/LGRS.2007.896328.
+    """
+
+    preds: List[Tensor]
+    target: List[Tensor]
+    higher_is_better: bool = False
+
```
Comment (SkafteNicki) [inline torchmetrics/image/ergas.py:62]: is the metric differentiable?
```suggestion
    higher_is_better: bool = False
    is_differentiable: bool = False

```

**Review by SkafteNicki — APPROVED** [review]
LGTM :]

File: docs/source/references/functional.rst [hunk 2]
```diff
@@ -270,13 +270,21 @@ peak_signal_noise_ratio [func]
...
 
+
 universal_image_quality_index [func]
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. autofunction:: torchmetrics.functional.universal_image_quality_index
     :noindex:
 
 
+error_relative_global_dimensionless_synthesis [func]
```
[the remark this answers is not in the document]
Comment (vumichien) [reply docs/source/references/functional.rst]: okay I got it

File: torchmetrics/image/ergas.py (hunk 1 above)
Comment (vumichien) [reply torchmetrics/image/ergas.py]: yes, this metric is differentiable

File: tests/helpers/reference_metrics.py [hunk 3]
```diff
@@ -202,6 +202,28 @@ def _calibration_error(
     return loss
 
 
+def _sk_ergas(preds, target, ratio, reduction):
```
Comment (SkafteNicki) [inline tests/helpers/reference_metrics.py:205]:
```suggestion
def _sk_ergas(
  preds: torch.Tensor, 
  target: torch.Tensor, 
  ratio: Union[int, float] = 4, 
  reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean"
) -> torch.Tensor:
  """ Reference implementation of Erreur Relative Globale Adimensionnelle de Synthèse """
```

File: tests/helpers/reference_metrics.py [hunk 4]
```diff
@@ -202,6 +202,28 @@ def _calibration_error(
...
+    if reduction == "sum":
+        to_return = torch.sum(ergas_score)
+    elif reduction == "elementwise_mean":
+        to_return = torch.mean(ergas_score)
+    else:
+        to_return = ergas_score
+    return to_return
+
+
 def _sk_sam(preds, target, reduction):
```
Comment (SkafteNicki) [inline tests/helpers/reference_metrics.py:227]:
```suggestion
def _sk_sam(
  preds: torch.Tensor, 
  target: torch.Tensor, 
  reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean"
) -> torch.Tensor:
  """ Reference implementation of spectral angle mapper. """
```

File: torchmetrics/functional/image/ergas.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    preds = preds.reshape(B, C, H * W)
+    target = target.reshape(B, C, H * W)
+
+    diff = preds - target
+    sum_squared_error = torch.sum(diff * diff, dim=2)
+    rmse_per_band = torch.sqrt(sum_squared_error / (H * W))
+    mean_target = torch.mean(target, dim=2)
+
+    ergas_score = 100 * ratio * torch.sqrt(torch.sum((rmse_per_band / mean_target) ** 2, dim=1) / C)
+    return reduce(ergas_score, reduction)
```
Comment (SkafteNicki) [inline torchmetrics/functional/image/ergas.py:82]: lets follow pep8 style and use lower case for variable names
```suggestion
    b, c, h, w = preds.shape
    preds = preds.reshape(b, c, h * w)
    target = target.reshape(b, c, h * w)

    diff = preds - target
    sum_squared_error = torch.sum(diff * diff, dim=2)
    rmse_per_band = torch.sqrt(sum_squared_error / (h * w))
    mean_target = torch.mean(target, dim=2)

    ergas_score = 100 * ratio * torch.sqrt(torch.sum((rmse_per_band / mean_target) ** 2, dim=1) / c)
    return reduce(ergas_score, reduction)
```

File: tests/helpers/reference_metrics.py [hunk 6]
```diff
@@ -202,6 +202,28 @@ def _calibration_error(
...
+    B, C, H, W = preds.shape
+    sk_preds = preds.reshape(B, C, H * W)
+    sk_target = target.reshape(B, C, H * W)
+    # compute rmse per band
+    diff = sk_preds - sk_target
+    sum_squared_error = torch.sum(diff * diff, dim=2)
+    rmse_per_band = torch.sqrt(sum_squared_error / (H * W))
+    mean_target = torch.mean(sk_target, dim=2)
+    # compute ergas score
+    ergas_score = 100 * ratio * torch.sqrt(torch.sum((rmse_per_band / mean_target) ** 2, dim=1) / C)
```
Comment (SkafteNicki) [inline tests/helpers/reference_metrics.py:216]: again lets follow pep8 standards
```suggestion
    b, c, h, w = preds.shape
    sk_preds = preds.reshape(b, c, h * w)
    sk_target = target.reshape(b, c, h * w)
    # compute rmse per band
    diff = sk_preds - sk_target
    sum_squared_error = torch.sum(diff * diff, dim=2)
    rmse_per_band = torch.sqrt(sum_squared_error / (h * w))
    mean_target = torch.mean(sk_target, dim=2)
    # compute ergas score
    ergas_score = 100 * ratio * torch.sqrt(torch.sum((rmse_per_band / mean_target) ** 2, dim=1) / c)
```

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
@vumichien mind check the:
```
E       b, c, h, w = preds.shape
E       ^
E   IndentationError: unexpected indent
```

**Comment by vumichien** [comment]
@SkafteNicki Thanks for helping me a lot

File: tests/image/test_ergas.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    (13, 3, 0.8, 2, torch.float32),
+    (14, 1, 0.7, 3, torch.double),
+    (15, 3, 0.5, 4, torch.float64),
+]:
+    preds = torch.rand(NUM_BATCHES, BATCH_SIZE, channel, size, size, dtype=dtype)
+    _inputs.append(
+        Input(
+            preds=preds,
+            target=preds * coef,
+            ratio=ratio,
```
Comment (Borda) [inline tests/image/test_ergas.py:42]:
```suggestion
            ratio=ratio
```

File: torchmetrics/functional/image/ergas.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            f" Got preds: {preds.shape} and target: {target.shape}."
+        )
+    return preds, target
+
+
+def _ergas_compute(
+    preds: Tensor,
+    target: Tensor,
+    ratio: Union[int, float] = 4,
+    reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean",
```
Comment (Borda) [inline torchmetrics/functional/image/ergas.py:51]:
```suggestion
    reduction: Literal["elementwise_mean", "sum", "none", None] = "elementwise_mean",
```

File: torchmetrics/functional/image/ergas.py [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+    Args:
+        preds: estimated image
+        target: ground truth image
+        ratio: ratio of high resolution to low resolution (default=4)
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
```
Comment (Borda) [inline torchmetrics/functional/image/ergas.py:63]:
```suggestion
            - ``'none'`` or ``None``: no reduction will be applied
```

File: torchmetrics/functional/image/ergas.py [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+    ergas_score = 100 * ratio * torch.sqrt(torch.sum((rmse_per_band / mean_target) ** 2, dim=1) / c)
+    return reduce(ergas_score, reduction)
+
+
+def error_relative_global_dimensionless_synthesis(
+    preds: Tensor,
+    target: Tensor,
+    ratio: Union[int, float] = 4,
+    reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean",
```
Comment (Borda) [inline torchmetrics/functional/image/ergas.py:89]:
```suggestion
    reduction: Literal["elementwise_mean", "sum", "none", None] = "elementwise_mean",
```

File: torchmetrics/functional/image/ergas.py [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+    Args:
+        preds: estimated image
+        target: ground truth image
+        ratio: ratio of high resolution to low resolution (default=4)
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
```
Comment (Borda) [inline torchmetrics/functional/image/ergas.py:101]:
```suggestion
            - ``'none'`` or ``None``: no reduction will be applied
```

File: torchmetrics/image/ergas.py [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+    image considering normalized average error of each band of the result image
+    (ErrorRelativeGlobalDimensionlessSynthesis).
+
+    Args:
+        ratio: ratio of high resolution to low resolution
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
```
Comment (Borda) [inline torchmetrics/image/ergas.py:37]:
```suggestion
            - ``'none'`` or ``None``: no reduction will be applied
```

File: torchmetrics/image/ergas.py [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+    preds: List[Tensor]
+    target: List[Tensor]
+    higher_is_better: bool = False
+    is_differentiable: bool = True
+
+    def __init__(
+        self,
+        ratio: Union[int, float] = 4,
+        reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean",
```
Comment (Borda) [inline torchmetrics/image/ergas.py:67]:
```suggestion
        reduction: Literal["elementwise_mean", "sum", "none", None] = "elementwise_mean",
```

**Review by Borda — APPROVED** [review]

**Pull request merged**